### DISCUSSION TITLE
Should WindowsToolchain set '/X' MSVC flag?

### DISCUSSION THREAD

**Discussion by redbaron** [MarkSchofield/Toolchain#111] [Q&A] [answered]
It feels like it would be safer to set "/X" to guarantee that nothing from env vars  (like VS command prompt or CLion on Windows) leaks into build process.

**Comment by MarkSchofield** [comment, marked as answer]
That’s a great idea. I wasn’t aware of that option.

**Reply by redbaron** [reply]
Sadly I didn't find anything like that for linker's `LIB` env var. In theory CMake allows `unset($ENV{LIB})` , but IDK if it just clears `$ENV` or it truly propagates to spawned processes like compilers and linkers.

**Reply by redbaron** [reply]
This way to clear `LIB`  env vars works for me:

```
set(CMAKE_CXX_LINKER_LAUNCHER ${CMAKE_COMMAND} -E env --unset=LIB --unset=LIBPATH --)
```
